Source/Scene/ImplicitSubtree.js

```diff
@@ -496,3 +602,52 @@ function parseAvailabilityBitstream(
     availableCount: availabilityJson.availableCount,
   });
 }
+
+/**
+ * Parse the 3DTILES_metadata table, storing a {@link MetadataTable} in the
+ * subtree.
+ *
+ * @param {ImplicitSubtree} subtree The subtree
+ * @param {Object} metadataExtension The JSON for the 3DTILES_metadata extension
+ * @param {ImplicitTileset} implicitTileset The implicit tileset this subtree belongs to.
+ * @param {Object} bufferViewsU8 A dictionary of bufferView index to its Uint8Array contents.
+ * @private
+ */
+function parseMetadataTable(
+  subtree,
+  metadataExtension,
+  implicitTileset,
+  bufferViewsU8
+) {
+  var tileCount = subtree._tileAvailability.availableCount;
+  var metadataClassName = metadataExtension.class;
+  var metadataSchema = implicitTileset.tileset.metadata.schema;
+  var metadataClass = metadataSchema.classes[metadataClassName];
+
+  subtree._metadataTable = new MetadataTable({
+    class: metadataClass,
+    count: tileCount,
+    properties: metadataExtension.properties,
+    bufferViews: bufferViewsU8,
+  });
+}
+
+/**
+ * Make a jump buffer, i.e. a map of tile bit index to the metadata entity ID.
+ * This is stored in the subtree.
+ *
+ * @param {ImplicitSubtree} subtree The subtree
+ * @private
+ */
+function makeJumpBuffer(subtree) {
+  var tileAvailability = subtree._tileAvailability;
+  var entityId = 0;
+  var jumpBuffer = {};
```

Comment:
It might be more efficient to implement the jump buffer as a typed array. And then somehow indicate which elements are no data (maybe a sentinel value, or maybe the tile availability buffer is good enough for that)

Reply:
okay, I guess I'll do something similar to what @IanLilleyT did, use a -1 as a sentinel

Reply:
Actually, looking closer at this, it's probably easier to manage if we just use tile availability. I'll look closer at this tomorrow.